### DISCUSSION TITLE
【tree】how to realize elliptical style when the custom treenode name is  too long

### DISCUSSION THREAD

**Discussion by hanyyy5** [ant-design/ant-design#31879] [Q&A] [answered]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
every treenode needs occupy the remain space of a line, there are two operational button in the end of the line, so the title props of treenode is customized；need to realize elliptical style when the custom treenode name is  too long

场景：<1>需要树节点占满一行（每一行最末端带两个操作按钮），故修改节点外层容器ant-tree-node-content-wrapper为flex:1, 同时设置display：flex;   
           <2>在自定义treeNode节点title的情况下，又设置了节点容器ant-tree-title为flex:1; 
           <3> 节点名称的文字过长需要超出省略,  由于外层设置了flex:1,  节点文字过长时直接撑开了外层节点容器ant-tree-title，无法超出省略；
           尝试将节点容器ant-tree-title的width设置为0，可以超出省略，但是展开收起节点时有个明显的卡顿感觉，应该是宽度从0开始变化引起的，所以应该如何实现节点超出省略？

### What does the proposed API look like?
are there any props can do this

**Comment by afc163** [comment]
> 需要树节点占满一行

有个 blockNode 可以试试。

**Reply by hanyyy5** [reply]
看了下blockNode应该是设置了节点外层wrapper的样式为flex:auto;  这个还是会被内部子元素撑开，子元素设置为nowrap不换行的时候，其内容有多长，这个节点就会被撑多长

**Comment by hanyyy5** [comment, marked as answer]
设置每行节点外层容器(类名为ant-tree-node)为固定宽度即可
